source/css/sass/mixins/_spacing.scss

```diff
@@ -93,149 +93,35 @@
-    @if $level == "extra-small" {
-      @include padding($baselinegrid-space-extra-small, $dimension);
-    } @else if $level == "small" {
-      @include padding($baselinegrid-space-small, $dimension);
-    } @else if $level == "smallish" {
-      @include padding($baselinegrid-space-smallish, $dimension);
-    } @else if $level == "medium" {
-      @include padding($baselinegrid-space-medium, $dimension);
-    } @else if $level == "large" {
-      @include padding($baselinegrid-space-large, $dimension);
-    }
-  } @else if $type == "margin" {
-    @if $level == "extra-small" {
-      @include margin($baselinegrid-space-extra-small, $dimension);
-    } @else if $level == "small" {
-      @include margin($baselinegrid-space-small, $dimension);
-    } @else if $level == "smallish" {
-      @include margin($baselinegrid-space-smallish, $dimension);
-    } @else if $level == "medium" {
-      @include margin($baselinegrid-space-medium, $dimension);
-    } @else if $level == "large" {
-      @include margin($baselinegrid-space-large, $dimension);
-    }
-  }
-}
-
-@mixin blg-pad-top--small-to-medium {
-
-  @include blg-spacing("block-start", "small");
-
-  @include mq($from: medium) {
-    @include blg-spacing("block-start", "medium");
-  }
-
-}
-
-@mixin blg-pad-bottom--small-to-medium {
-
-  @include blg-spacing("block-end", "small");
-
-  @include mq($from: medium) {
-    @include blg-spacing("block-end", "medium");
-  }
-
-}
-
-@mixin blg-pad-top--large-to-extra-large {
-
-  @include blg-spacing("block-start", "large");
-
-  @include mq($from: medium) {
-    @include blg-spacing("block-start", "extra-large");
-  }
-
-}
-
-@mixin blg-pad-bottom--large-to-extra-large {
-
-  @include blg-spacing("block-end", "large");
-
-  @include mq($from: medium) {
-    @include blg-spacing("block-end", "extra-large");
-  }
-
-}
-
-@mixin blg-pad-vertical-small-to-medium {
-  @include blg-pad-top--small-to-medium();
-  @include blg-pad-bottom--small-to-medium();
-}
-
-@mixin blg-pad-vertical-large-to-extra-large {
-  @include blg-pad-top--large-to-extra-large();
-  @include blg-pad-bottom--large-to-extra-large();
-}
-
-@mixin blg-margin-bottom--medium-to-large {
-
-  @include blg-spacing("block-end", "medium", "margin");
-
-  @include mq($from: medium) {
-    @include blg-spacing("block-end", "large", "margin");
-  }
-
-}
-
-@mixin blg-margin-bottom--small-to-medium {
-
-  @include blg-spacing("block-end", "small", "margin");
-
-  @include mq($from: medium) {
-    @include blg-spacing("block-end", "medium", "margin");
-  }
-
-}
-
-@mixin blg-column-container() {
-  @include blg-margin-bottom--medium-to-large();
+@mixin block-spacing($end: $baselinegrid-space-small, $start: 0) {
+  @include margin($start $end, block);
 }
 
 @mixin h1-spacing() {
-  margin: 0;
+  @include block-spacing();
 }
 
 @mixin h2-spacing() {
-  margin: 0;
-  @include padding(21px, "block-end");
-  @include padding(21px, "block-start");
+  @include block-spacing();
```

Comment:
That's fine, but we should make it `calc(#{$baselinegrid-space-small} - 1px)`.

Reply:
`s/- 1px/- 3px/`?

Tried that approach originally and it doesn't scale completely correctly when changing the baseline grid size. That said it's better than this fixed `px` as it is at the moment as we're building for the general case, so let's do it. Probably prefer a sass calculation as no browser-side moving parts.

Reply:
Mistook it for the content header divider, which is a `1px` thing.

There should be a way to calculate what these values are, `3px` is a magic number otherwise...

Reply:
After some experimentation in https://type-scale.com/, looks like we're using the Major Second (1.125) scale, which should help turn these into calculations.

Reply:
First attempt in d924e43.

Reply:
What looks like browser rounding errors knock things off the scale a bit, but I think we can live with it. As an expression of intent it works well, and it removes the previously magic numbers we had, and makes the baseline grid more easily configurable.